.github/workflows/build-ubuntu.yaml

```diff
@@ -0,0 +1,59 @@
+name: build-ubuntu
+
+
+# Controls when the action will run
+on:
+  push:
+  pull_request:
+  workflow_dispatch: # trigger the workflow manually
+  # # trigger the workflow on schedule
+  # # https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#scheduled-events
+  # schedule:
+  #   - cron:  '* * * * *'
+
+
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  build:
+    runs-on: ${{ matrix.runner }}
+    continue-on-error: false
+    strategy:
+      matrix:
+        runner: [ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
```

Comment:
Are we really running 12 tests here, seems excessive :)

Skip Ubuntu 16.04, that one is effectively dead and it would require different cmake since it comes with 3.5 and we require 3.10.